docs/apis/streaming/windows.md

```diff
@@ -24,1023 +24,608 @@ specific language governing permissions and limitations
-             one window (since windows overlap by at most 4 seconds)
-             The notion of time is specified by the selected TimeCharacteristic (see <a href="{{ site.baseurl }}/apis/streaming/event_time.html">time</a>).
-      {% highlight scala %}
-keyedStream.timeWindow(Time.seconds(5), Time.seconds(1))
-      {% endhighlight %}
-            </p>
-          </td>
-        </tr>
-      <tr>
-        <td><strong>Tumbling count window</strong><br>KeyedStream &rarr; WindowedStream</td>
-        <td>
-          <p>
-          Defines a window of 1000 elements, that "tumbles". This means that elements are
-          grouped according to their arrival time (equivalent to processing time) in groups of 1000 elements,
-          and every element belongs to exactly one window.
-    {% highlight scala %}
-keyedStream.countWindow(1000)
-    {% endhighlight %}
-        </p>
-        </td>
-      </tr>
-      <tr>
-      <td><strong>Sliding count window</strong><br>KeyedStream &rarr; WindowedStream</td>
-      <td>
-        <p>
-          Defines a window of 1000 elements, that "slides" every 100 elements. This means that elements are
-          grouped according to their arrival time (equivalent to processing time) in groups of 1000 elements,
-          and every element can belong to more than one window (as windows overlap by at most 900 elements).
-  {% highlight scala %}
-keyedStream.countWindow(1000, 100)
-  {% endhighlight %}
-        </p>
-      </td>
-    </tr>
-  </tbody>
-</table>
-
+input
+    .keyBy(<key selector>)
+    .window(<window assigner>)
+    .<windowed transformation>(<window function>)
+{% endhighlight %}
 </div>
 </div>
 
-### Advanced Window Constructs
+We will cover [window assigners](#window-assigners) in a separate section below.
+
+The window transformation can be one of `reduce()`, `fold()` or `apply()`. Which respectively
+takes a `ReduceFunction`, `FoldFunction` or `WindowFunction`. We describe each of these ways
+of specifying a windowed transformation in detail below: [window functions](#window-functions).
+
+For more advanced use cases you can also specify a `Trigger` that determines when exactly a window
+is being considered as *ready for processing*. These will be covered in more detail in
+[triggers](#triggers).
+
+## Window Assigners
+
+The window assigner specifies how elements of the stream are divided into finite slices. Flink comes
+with pre-implemented window assigners for the most typical use cases, namely *tumbling windows*,
+*sliding windows*, *session windows* and *global windows* but you can implement your own by
+extending the `WindowAssigner` class. All the built-in window assigners, except for the global
+windows one, assign elements to windows based on time, which can either be processing time or event
+time.
+
+Let's first look at how each of these window assigners works before looking at how they can be used
```

Comment:
I thought it was with "s" because each of the window assigners is a singular. Could be wrong, though.